lib/vcr/cassette.rb

```diff
@@ -149,8 +149,13 @@ def recording?
     def serializable_hash
       {
         "http_interactions" => interactions_to_record.map(&:to_hash),
-        "recorded_with"     => "VCR #{VCR.version}"
-      }
+        "recorded_with"     => "VCR #{VCR.version}",
+        "metadata"          => metadata
+      }.reject { |k, v| v.empty? }
+    end
+
+    def metadata
+      @metadata ||= recording? ? {} : deserialized_hash['metadata'] 
```

Comment:
Perhaps we require something like this?

```suggestion
      @metadata ||= recording? ? {} : (deserialized_hash['metadata'] || {})
```

There already existed tests which had a `nil` value in metadata, so we had coverage of that branch of things.